nv/decode.go

```
@@ -91,22 +102,29 @@ func decodeList(dec decoder, target reflect.Value) error {
 		var targetField reflect.Value
 		if !isMap {
 			targetFieldIndex, ok := targetFieldIndexMap[pName]
-			// If there's no corresponding struct field, skip the data and move
-			// on to the next data pair
+			// If there's no corresponding struct field and no field for
+			// extras, skip the data and move on to the next data pair. If
+			// there is an field for extras, make that the new target for this
+			// iteration.
 			if !ok {
-				if err := dec.skip(); err != nil {
-					return err
+				if !extraMap.IsValid() {
```

I would return an error from `fieldIndexMap()` and bubble that up. Just `extra` or error on name tag collisions too? Error on both seems the way to go.